src/ui/pages/AssetInfo/AssetAddressDetails.tsx

```diff
@@ -0,0 +1,740 @@
+import React, { useRef, useState } from 'react';
+import { HStack } from 'src/ui/ui-kit/HStack';
+import { TokenIcon } from 'src/ui/ui-kit/TokenIcon';
+import { UIText } from 'src/ui/ui-kit/UIText';
+import LinkIcon from 'jsx:src/ui/assets/new-window.svg';
+import ChainsIcon from 'jsx:src/ui/assets/pie-chart.svg';
+import {
+  formatCurrencyToParts,
+  formatCurrencyValue,
+} from 'src/shared/units/formatCurrencyValue';
+import { middot } from 'src/ui/shared/typography';
+import { useCurrency } from 'src/modules/currency/useCurrency';
+import WalletIcon from 'jsx:src/ui/assets/wallet-fancy.svg';
+import EyeIcon from 'jsx:src/ui/assets/eye.svg';
+import ArrowLeftIcon from 'jsx:src/ui/assets/arrow-left.svg';
+import type { AssetFullInfo } from 'src/modules/zerion-api/requests/asset-get-fungible-full-info';
+import { VStack } from 'src/ui/ui-kit/VStack';
+import type { WalletAssetDetails } from 'src/modules/zerion-api/requests/wallet-get-asset-details';
+import { UnstyledAnchor } from 'src/ui/ui-kit/UnstyledAnchor';
+import { formatPercent } from 'src/shared/units/formatPercent/formatPercent';
+import { formatTokenValue } from 'src/shared/units/formatTokenValue';
+import { NeutralDecimals } from 'src/ui/ui-kit/NeutralDecimals';
+import { useWalletPortfolio } from 'src/modules/zerion-api/hooks/useWalletPortfolio';
+import { useHttpClientSource } from 'src/modules/zerion-api/hooks/useHttpClientSource';
+import { UnstyledButton } from 'src/ui/ui-kit/UnstyledButton';
+import ArrowDownIcon from 'jsx:src/ui/assets/caret-down-filled.svg';
+import type { ExternallyOwnedAccount } from 'src/shared/types/ExternallyOwnedAccount';
+import { isReadonlyAccount } from 'src/shared/types/validators';
+import type { HTMLDialogElementInterface } from 'src/ui/ui-kit/ModalDialogs/HTMLDialogElementInterface';
+import { WalletDisplayName } from 'src/ui/components/WalletDisplayName';
+import { WalletAvatar } from 'src/ui/components/WalletAvatar';
+import { UnstyledLink } from 'src/ui/ui-kit/UnstyledLink';
+import { Button } from 'src/ui/ui-kit/Button';
+import { CenteredDialog } from 'src/ui/ui-kit/ModalDialogs/CenteredDialog';
+import { useWalletAssetPnl } from 'src/modules/zerion-api/hooks/useWalletAssetPnl';
+import type { AssetAddressPnl } from 'src/modules/zerion-api/requests/asset-get-fungible-pnl';
+import { formatPriceValue } from 'src/shared/units/formatPriceValue';
+import { AssetHeader } from './AssetHeader';
+import { getColor, getSign } from './helpers';
+
+function Line() {
+  return (
+    <div
+      style={{
+        width: '100%',
+        height: 2,
+        backgroundColor: 'var(--neutral-200)',
+      }}
+    />
+  );
+}
+
+function LoadingSkeleton() {
+  return (
+    <div
+      style={{
+        width: 100,
+        height: 24,
+        backgroundColor: 'var(--neutral-200)',
+        borderRadius: 12,
+      }}
+    />
+  );
+}
```

Comment:
Came up with a slightly better design:
```suggestion
function LoadingSkeleton() {
  return (
    <div
      style={{
        width: 100,
        height: 24,
      }}
    />
  );
}
```